lib/oanda_api/client/client.rb

```diff
@@ -158,9 +170,11 @@ def self.last_throttled_at
     # @param [Time] time The time that the throttle was requested.
     #
     # @return [void]
-    def self._throttle(time)
-      @throttle_mutex.synchronize { @throttled_at = time }
-      sleep OandaAPI.configuration.min_request_interval
+    def self._throttle(delta, time)
+      @throttle_mutex.synchronize do
+        @throttled_at = time
+        sleep OandaAPI.configuration.min_request_interval - delta
```

Comment:
Sleeping while holding a lock sounds like a recipe for deadlock. If you put this thread to sleep while having the mutex locked, if the woken thread also needs to call #_throttle, it would be blocked waiting for the mutex to be released.  I see the only purpose of the mutex to be ensuring that the get/setting of attributes (@throttled_at, @last_request_at) is consistent, and thereby allowing safe access across all threads.